apps/bfd-pipeline/bfd-pipeline-rda-grpc/src/main/java/gov/cms/bfd/pipeline/rda/grpc/DcGeoRDALoadJob.java

```diff
@@ -0,0 +1,149 @@
+
+/**
+ * Skeleton PipelineJob instance that delegates the actual ETL work to two other objects. The
+ * RDASource object handles communication with the source of incoming data. The RDASink object
+ * handles communication with the ultimate storage system. The purpose of this class is to handle
+ * general PipelineJob semantics that are common to any source or sink.
+ */
+public final class DcGeoRDALoadJob implements PipelineJob {
+  private static final Logger LOGGER = LoggerFactory.getLogger(DcGeoRDALoadJob.class);
+  public static final String SCAN_INTERVAL_PROPERTY = "DCGeoRDALoadIntervalSeconds";
+  public static final String RUN_TIME_PROPERTY = "DCGeoRDALoadRunSeconds";
+  public static final String MAX_RECORDS_PROPERTY = "DCGeoRDALoadMaxRecords";
+  public static final int SCAN_INTERVAL_DEFAULT = 300;
+  public static final int RUN_TIME_DEFAULT = 300;
+  public static final int MAX_RECORDS_DEFAULT = Integer.MAX_VALUE;
+  public static final String BATCH_SIZE_PROPERTY = "DCGeoBatchSize";
+  public static final int BATCH_SIZE_DEFAULT = 1;
+  public static final String CALLS_METER_NAME =
+      MetricRegistry.name(DcGeoRDALoadJob.class.getSimpleName(), "calls");
+  public static final String FAILURES_METER_NAME =
+      MetricRegistry.name(DcGeoRDALoadJob.class.getSimpleName(), "failures");
+  public static final String SUCCESSES_METER_NAME =
+      MetricRegistry.name(DcGeoRDALoadJob.class.getSimpleName(), "successes");
+  public static final String PROCESSED_METER_NAME =
+      MetricRegistry.name(DcGeoRDALoadJob.class.getSimpleName(), "processed");
+
+  private final Config config;
+  private final Callable<RDASource<PreAdjudicatedClaim>> sourceFactory;
+  private final Callable<RDASink<PreAdjudicatedClaim>> sinkFactory;
+  private final Meter callsMeter;
+  private final Meter failuresMeter;
+  private final Meter successesMeter;
+  private final Meter processedMeter;
+
+  public DcGeoRDALoadJob(
+      Config config,
+      Callable<RDASource<PreAdjudicatedClaim>> sourceFactory,
+      Callable<RDASink<PreAdjudicatedClaim>> sinkFactory,
+      MetricRegistry appMetrics) {
+    this.config = config;
+    this.sourceFactory = sourceFactory;
+    this.sinkFactory = sinkFactory;
+    callsMeter = appMetrics.meter(CALLS_METER_NAME);
+    failuresMeter = appMetrics.meter(FAILURES_METER_NAME);
+    successesMeter = appMetrics.meter(SUCCESSES_METER_NAME);
+    processedMeter = appMetrics.meter(PROCESSED_METER_NAME);
+  }
+
+  /**
+   * Factory method to construct a new job instance using standard parameters.
+   *
+   * @param appMetrics MetricRegistry used to track operational metrics
+   * @return a DcGeoRDALoadJob instance suitable for use by PipelineManager.
+   */
+  public static DcGeoRDALoadJob newDcGeoRDALoadJob(MetricRegistry appMetrics) {
+    return new DcGeoRDALoadJob(
+        new Config(),
+        () -> new SkeletonRDASource(appMetrics),
+        () -> new SkeletonRDASink(appMetrics),
+        appMetrics);
+  }
```

Comment:
Interesting style. Why a `static` method instead of just an overloaded-and-chained constructor?

Reply:
Just a style thing. I think it helps maintainability a bit. When the object will be constructed by a call from somewhere outside the module I like a static method since it decouples the constructor to use. It also avoids having too many constructors.. But it's not a major thing. Could go the overloaded constructor route too.

Reply:
To be clear, I don't have a strong opinion -- was just curious.